### ISSUE TITLE
All numeric encoded hashids being treated as natural IDs

### ISSUE THREAD

**Issue by afshelley** [nshafer/django-hashid-field#66]
Hi, I have run into an issue whereby an object has a fully-numeric coded hashid. Despite a comment in the library asserting this is not possible I have found that it is. Here are the steps to recreate:

```
import hashids
a = hashids.Hashids("salt", min_length=7, alphabet="0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ")
a.encode(8308830)
>>> '5666668'
```

I realise now that this may have been avoided if we used a prefix, but unfortunately we did not.

I think the issue is that the `_is_uint()` method used by the `Hashid` class assumes if a value can be cast to an unsigned integer, that it needs to be _encoded_ rather than _decoded_. This means when we are deserialising a numeric hashid to try and resolve a foreign key relation (we are using DRF) the numeric hashid is being treated as a natural ID, and the wrong object is queried.

I'm not sure what the solution for this is. Would it be possible to add a setting e.g. `HASHID_FIELD_RESPECT_TYPE` or something (defaults to false) and then write the `_is_unit()` method like this:

```
from .conf import settings

def _is_uint(number):
    """Returns whether a value is an unsigned integer."""
    if settings.HASHID_FIELD_RESPECT_TYPE:
        return isinstance(number, int) and int(number) >= 0
    try:
        return number == int(number) and number >= 0
    except ValueError:
        return False
```

**Comment by nshafer** [comment]
Thank you for the detailed report. Assuming that all hashids would have non-numeric characters in it was definitely a falsehood.

Since this is actually a bug introduced in 3.2.0 (and I'm surprised nobody else has hit it), I've fixed it straightaway without a feature flag and released 3.3.2.

A lot of code had been changed in 3.2.0 when prefix support was added, and this was due to an optimization related to it. However, I did perf tests, and the fix I just published in 3.3.2 has no significant performance implication.

I also added tests for these digit-only hashids to hopefully prevent another regression.

Thanks!
Nate
